test/integration/spec/connect.js

```diff
@@ -203,6 +204,119 @@ describe('connect', function() {
+
+      let scenario = 'when called ';
+      if (isInvalidRegion) {
+        scenario += `with an invalid mediaRegion : ${mediaRegion}`;
+      } else if (mediaRegion === 'without') {
+        scenario += 'without a mediaRegion ';
+      } else {
+        scenario += `with a valid mediaRegion: ${mediaRegion}`;
+      }
+
+      context(scenario, () => {
+        it(`should return a CancelablePromise that ${isInvalidRegion ? 'rejects with a MediaServerRemoteDescFailedError if the room is not p2p' : 'resolves with a Room'}`, async () => {
+          const cancelablePromise = connect(token, Object.assign({ name: sid }, mediaRegionOptions, { tracks: [] }));
+          assert(cancelablePromise instanceof CancelablePromise);
+
+          let error = null;
+          let room = null;
+          try {
+            room = await cancelablePromise;
+          } catch (error_) {
+            error = error_;
+          } finally {
+            if (room) {
+              room.disconnect();
+              await completeRoom(sid);
+            }
+            if (isInvalidRegion) {
+              if (defaults.topology !== 'peer-to-peer') {
+                assert(error instanceof MediaServerRemoteDescFailedError, `Connected to Room ${room && room.sid} with an invalid mediaRegion "${mediaRegion}"`);
+              } else {
+                assert.equal(room.mediaRegion, null);
+              }
+            } else {
+              assert.equal(error, null);
+              if (defaults.topology === 'peer-to-peer') {
+                assert.equal(room.mediaRegion, null);
+              }
+              if (defaults.topology !== 'peer-to-peer') {
+                if (['without'].includes(mediaRegion)) {
+                  assert.equal(typeof room.mediaRegion, 'string');
+                } else {
+                  assert.equal(room.mediaRegion, mediaRegion);
+                }
+              }
+              assert(room instanceof Room);
+            }
+          }
+        });
+      });
+    });
+
+    if (defaults.topology !== 'peer-to-peer') {
+      it('should allow the first participant to set the Room\'s media region', async () => {
+        let room;
+        let roomRegion;
+        const cancelablePromise = connect(token, Object.assign({ name: sid }, { mediaRegion: 'us2' }, defaults, { tracks: [] }));
+
+        try {
+          room = await cancelablePromise;
+          roomRegion = room.mediaRegion;
+          roomName = room.name;
+          assert.equal(roomRegion, 'us2');
+        } catch (error) {
+          roomRegion = room.mediaRegion;
+          assert.throws(assert.equal(error, roomRegion));
+          room.disconnect();
+        } finally {
+          if (room) {
+            room.disconnect();
+          }
+        }
+      });
+
+      it('should have another participant match the media region of the first participant regardless of the parameter being passed in', async () => {
```

Comment:
This test can re-use the first participant in the previous test.

Reply:
Reuse the first participant from the tests ran in the context block? Do you think I should just move the entire remote participant it block into the context block?